### PR TITLE
fix(StringQuotingEmitter): add missing patterns for double quoting

### LINKED ISSUE #953: Yaml Serialization is missing some tokens that need to be double quoted
Hello,

I found out that there are some special characters that are not being considered to be double quoted when serializing a Kubernetes object to yaml.
In the **StringQuotingEmitter** class, there is this regex:

`private static readonly Regex QuotedRegex =
            new Regex(@"^(\~|null|true|false|-?(0|[0-9]*)(\.[0-9]*)?([eE][-+]?[0-9]+)?)?$");`

This is not taking into consideration, at least, the following values "True", "TRUE", "False", "FALSE", "Y", "N", "y", "n", "Null" and "NULL".

### PR DESCRIPTION
**Pull request by vitorpmoreira**
Fixes #953  
**Added the missing patterns to the StringQuotingEmmiter** class to ensure that these values are double quoted when serializing to yaml.
Added a complementary Unit Test.

### CODE REVIEW

**Comment by tg123** [comment]
should be https://yaml.org/type/bool.html right

**Comment by vitorpmoreira** [comment]
> 

yes, nice one! And the null is also missing the "~" as in https://yaml.org/type/null.html

File: src/KubernetesClient.Models/StringQuotingEmitter.cs [hunk 1]
```diff
@@ -10,7 +10,7 @@ public class StringQuotingEmitter : ChainedEventEmitter
     {
         // Patterns from https://yaml.org/spec/1.2/spec.html#id2804356
```
Comment (tg123) [inline src/KubernetesClient.Models/StringQuotingEmitter.cs:11]: could you please also add `https://yaml.org/type/bool.html`

seems no `on/off` in spec 1.2

Comment (vitorpmoreira) [reply src/KubernetesClient.Models/StringQuotingEmitter.cs:11]: sure! by any chance, is there any documentation regarding the spec versions that are supported by the k8s yaml parser? shouldn't they be the same for all client libraries + kubectl itself?

**Review by tg123 — APPROVED** [review]
/LGTM

**Pull request merged**